File: packages/java/src/com/peruspatial/ubigeo/IsValidUbigeo.java
package com.peruspatial.ubigeo;

public final class IsValidUbigeo {
  private IsValidUbigeo() {}

  public static boolean isValidUbigeo(String code) {
    if (code == null) return false;
    if (!code.chars().allMatch(Character::isDigit) || code.isEmpty()) return false;
    return switch (code.length()) {
      case 2 -> GetDepartment.getDepartment(code) != null;
      case 4 -> GetProvince.getProvince(code) != null;
      case 6 -> GetDistrict.getDistrict(code) != null;
      default -> false;
    };
  }
}

File: packages/rust/src/ubigeo/is_valid_ubigeo.rs
use super::get_department::get_department;
use super::get_district::get_district;
use super::get_province::get_province;

pub fn is_valid_ubigeo(code: &str) -> bool {
    if !code.bytes().all(|b| b.is_ascii_digit()) {
        return false;
    }
    match code.len() {
        2 => get_department(code).is_some(),
        4 => get_province(code).is_some(),
        6 => get_district(code).is_some(),
        _ => false,
    }
}
